### PR TITLE
sdjournal: use internal dlopen package

### PR DESCRIPTION
**Pull request by saschagrunert**
The first step for go modules as discussed here: https://github.com/coreos/go-systemd/pull/307#issuecomment-548800039

cc @lucab 

Afterwards I would need assistance in releasing v21 :innocent:

### CODE REVIEW

File: internal/dlopen/dlopen_example.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2015 CoreOS, Inc.
```
Comment (lucab) [inline internal/dlopen/dlopen_example.go]: We can skip this example file entirely.

Comment (saschagrunert) [reply internal/dlopen/dlopen_example.go]: Yes, I removed it.

**Comment by saschagrunert** [comment]
> A minor thing, plus `Gopkg.toml` needs to be updated too. Otherwise looks fine.

Thanks for the review, I'll updated `Gopkg.tomll` too.

**Comment by lucab** [comment]
@saschagrunert  https://github.com/coreos/go-systemd/releases/tag/v21

**Pull request merged**